### ISSUE TITLE
Simplify `Result<()>` uses and implement a `checkpatch.pl` check for it

### ISSUE THREAD

**Issue by ojeda** [Rust-for-Linux/linux#1128]
There are 5 cases of `Result<()>` at the time of writing -- can they be simplified?

Additionally, implement a `checkpatch.pl` check for it. This should be done in a different patch.

Please note that the `checkpatch.pl` maintainers will need to agree to the change.

---

This requires submitting a proper patch to the LKML and the Rust for Linux mailing list. Please recall to test your changes (including generating the documentation if changed, running the Rust doctests if changed, etc.), to use a proper title for the commit, to sign your commit under the Developer's Certificate of Origin and to add a `Suggested-by:` tag and a `Link:` tag to this issue. Please see https://rust-for-linux.com/contributing for details.

Please take this issue only if you are new to the kernel development process and you would like to use it as a test to submit your first patch to the kernel. Please do not take it if you do not plan to make other contributions to the kernel.

**Comment by ojeda** [comment]
v1: https://lore.kernel.org/rust-for-linux/20241117-simplify-result-v1-1-5b01bd230a6b@iiitd.ac.in/
v2: https://lore.kernel.org/rust-for-linux/20241118-simplify-result-v2-0-9d280ada516d@iiitd.ac.in/
net: https://lore.kernel.org/rust-for-linux/20241118-simplify-result-qt2025-v1-1-f2d9cef17fca@iiitd.ac.in/

**Comment by weirdsmiley** [comment]
@ojeda I have the second patch for `checkpatch.pl` ready. But I have been holding it, as I haven't resolved the signature issue yet. Should I send it regardless for other feedbacks?

**Comment by ojeda** [comment]
Sure, that could be useful -- but I would recommend making it clear that you are working on that, e.g. with a "Not-yet-Signed-off-by" tag or similar, or a note below the first `---` line.

Having said that, I think we are still waiting on the `checkpatch.pl` maintainer(s) for a similar issue, so I am not sure how much time they have currently: https://lore.kernel.org/rust-for-linux/CANiq72noG77qLF1WcRu4qJNboyqs9Xvvu_ZmjVN8qitK_MNWnw@mail.gmail.com/

Thanks!

**Comment by ojeda** [comment]
https://lore.kernel.org/rust-for-linux/20241128-add-checkpatch-result-simplify-v1-1-973603191d71@iiitd.ac.in/

**Comment by nicoan** [comment]
Hello @ojeda! 

Is the `checkpatch.pl`  implementation still needed? (taking in account https://github.com/rust-lang/rust-clippy/issues/14848 and https://github.com/rust-lang/rust-clippy/pull/16123)  I see that the last patch for the `checkpatch.pl`  was sent in Nov 2024, I can resume work for this if it makes sense.

**Comment by ojeda** [comment]
Yeah, ideally those lints will be merged and cover this case, so that should be fine :)
